src/backend/distributed/planner/recursive_planning.c

```diff
@@ -316,6 +351,189 @@ RecursivelyPlanCTEs(Query *query, RecursivePlanningContext *planningContext)
 }
 
 
+/*
+ * RecursivelyPlanSubqueryWalker recursively finds all the Query nodes and
+ * recursively plans if necessary.
+ */
+static bool
+RecursivelyPlanSubqueryWalker(Node *node, RecursivePlanningContext *context)
+{
+	if (node == NULL)
+	{
+		return false;
+	}
+
+	if (IsA(node, Query))
+	{
+		Query *query = (Query *) node;
+		DeferredErrorMessage *error = NULL;
+
+		context->level += 1;
+		error = RecursivelyPlanSubqueriesAndCTEs(query, context);
```

Comment:
This can probably use some elaboration.

```c
/*
 * First, make sure any subqueries and CTEs within this subquery are recursively planned
 * if necessary.
 */
```
and then below maybe

```c
/*
 * Recursively plan this subquery if it cannot be pushed down and is eligible for recursive planning.
 */
```